internal/schemautil/userconfig/desc.go

```diff
@@ -5,23 +5,72 @@ import (
 	"strings"
 )
 
+// EntityType is a type that represents the type of an entity.
+type EntityType int
+
+const (
+	// Resource is a constant that represents the resource entity type.
+	Resource EntityType = iota
+	// DataSource is a constant that represents the data source entity type.
+	DataSource
+)
+
+// String is a function that returns the string representation of the entity type.
+func (et EntityType) String() string {
+	return [...]string{"resource", "datasource"}[et]
```

Comment:
```suggestion
	return [...]string{"resource", "data source"}[et]
```

Reply:
Done. Sorry about that, we call it `datasource` in the code :D